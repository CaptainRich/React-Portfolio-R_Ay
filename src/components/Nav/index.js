
import React, { useEffect } from 'react';
import { capitalizeFirstLetter } from "../../utils/helpers";

function Nav( props ) {


    // Destructure 'props' from what was set in 'app.js'
    const {
        options = [],
        setCurrentOption,
        currentOption,
        contactSelected,
        setContactSelected
    } = props;
    
      useEffect(() => {
        document.title = capitalizeFirstLetter(currentOption.name);
      }, [currentOption]);


    // const handleClick = (item) => {
    //     console.log(item);
    //     return item;
    //   };

    console.log("Options: ", options);

    return (
        <header className="flex-row px-1">
            <h2>
                <a data-testid="link" href="/">
                    <span role="img" aria-label="briefcase"> ðŸ’¼</span> My Portfolio
                </a>
            </h2>

            <nav>
                <ul className="flex-row">
                    {/* <li className={`mx-2 ${!contactSelected && 'navActive'}`}>
                        <a data-testid="about" href="#about" onClick={() => setContactSelected(false)} >
                            About me
                        </a>
                    </li> */}


                    {options.map((option) => (
                        <li
                            className={`mx-1 ${
                                currentOption.name === option.name && !contactSelected && 'navActive'}`}
                            key={option.name}
                        >
                            <span
                                onClick={() => {
                                    setCurrentOption(option);
                                    setContactSelected(false);
                                }}
                            >
                                {capitalizeFirstLetter(option.name)}
                            </span>
                        </li>
                    ))}                   

                    <li className={`mx-2 ${contactSelected && 'navActive'}`}>
                        <span onClick={() => setContactSelected(true)}>Contact</span>
                    </li>
                    
                </ul>
            </nav>
        </header>
    );
}

export default Nav;