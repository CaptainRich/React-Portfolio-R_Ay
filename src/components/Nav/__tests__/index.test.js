
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

// Add the data and mock functions to address the 'props' passed into 
// the Nav() function.
const options = [
  { name: 'about', description: 'A brief personal overview.' }
]
const mockCurrentOption = jest.fn();
const mockSetCurrentOption = jest.fn();



// Invoke the cleanup function, to remove components from the DOM to prevent
// memory leaks as well as variable/data collisions between tests.
afterEach( cleanup );

// Declare the component we're testing
describe( "About component", () =>{
    
        
    // First test - baseline to verify the component is rendering
    it('renders', () => {
      render(<Nav
        options={options}
        setCurrentCategory={mockSetCurrentOption}
        currentCategory={mockCurrentOption}
      />);
      });

    // Second test - compare a snapshot of the 'Nav' component
    it('matches snapshot', () => {
        
        const { asFragment } = render(<Nav 
          options={options}
          setCurrentOption={mockSetCurrentOption}
          currentOption={mockCurrentOption}
        />);
        
        expect(asFragment()).toMatchSnapshot();
      });

});

// Now test for the 'briefcase emoji'
describe('emoji is visible', () => {

    it('inserts emoji into the h2', () => {

        // Arrange
        const { getByLabelText } = render(<Nav 
          options={options}
          setCurrentOption={mockSetCurrentOption}
          currentOption={mockCurrentOption}
        />);

        // Assert  
        expect(getByLabelText('briefcase')).toHaveTextContent('ðŸ’¼');

    })
});


// Now test the navigation links
describe('links are visible', () => {

    it('inserts text into the links', () => {

      // Arrange
      const { getByTestId } = render(<Nav 
        options={options}
        setCurrentOption={mockSetCurrentOption}
        currentOption={mockCurrentOption}
        />);

      // Assert
      expect(getByTestId('link')).toHaveTextContent('Portfolio for Richard Ay');
      //expect(getByTestId('about')).toHaveTextContent('about');

    });
  });