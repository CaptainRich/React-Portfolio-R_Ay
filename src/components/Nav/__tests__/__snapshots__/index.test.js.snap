// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="briefcase"
          role="img"
        >
           ðŸ’¼
        </span>
         Portfolio for Richard Ay
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-1 false"
        >
          <span>
            About
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
